### PR TITLE
Added PlaybackOnlyAttribute

### PR DESCRIPTION
**Pull request by seanmcc-msft**
We would like the ability to mark some tests as playback-only.

For example: https://github.com/Azure/azure-sdk-for-net/pull/12609/files#diff-e152777102db76590298fbda43777754R1769

Related https://github.com/Azure/azure-sdk-for-net/pull/12609

### CODE REVIEW

**Comment by pakrym** [comment]
Why do you want to ever avoid running these tests live?

**Comment by pakrym** [comment]
Read through the test. Would we ever reenable it? 

If we ever add this attribute I think it needs to take a reason for playback only.

File: sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Attribute on test assemblies, classes, or methods that run only against playback resources.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly, AllowMultiple = true, Inherited = true)]
+    public class PlaybackOnlyAttribute : NUnitAttribute, IApplyToTest
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:15]: Needs a tests.

**Comment by seanmcc-msft** [comment]
> Why do you want to ever avoid running these tests live?

In the case of ORS, we would have to take a dependency on the Storage Management SDK to set up and tear down the ORS policy with every test run.  This turned out to be a non-viable solution, so instead, we are going to run the tests in playback only.  We've had a few other similar cases, where it's not worth the effort to set up the infrastructure to live test a minor feature.

@tg-msft recommended we create this annotation at one point. 

@kasobol-msft @amnguye

**Comment by seanmcc-msft** [comment]
Another example is https://github.com/Azure/azure-sdk-for-net/issues/12623.  BlockBlobStorage storage accounts are legency, you can't even create one in the Azure Portal.  I'm pretty sure its not possible to create one via ARM templates.  To sidestep this, we gain access to a BlockBlobStorage account, create the test, record it, and we're good.

File: sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs (hunk 1 above)
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:15]: Would this prevent someone from being able to re-record the test just by changing the ctor parameter?

File: sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    {
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:16]: Let's add a "reason" required ctor parameter.

**Comment by tg-msft** [comment]
I was thinking of something a little more like a `[ManualTest]` attribute using [the Test Category we already skip for pipeline runs](https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/jobs/archetype-sdk-tests.yml#L72) plus maybe an EnvVar so folks can always skip it locally in VS without thinking.  It would still be easy enough to run them live without making code changes to remove the attribute, etc.

**Comment by pakrym** [comment]
We can also skip the test dynamically based on the existence of particular environment setting.

**Comment by tg-msft** [comment]
As long as the env setting was present in our DevOps pipeline because I'm pretty sure those machines are refurbished raspberry pi zeros and struggle getting some of these longer tests to finish before our timeout kills the whole run.

**Comment by seanmcc-msft** [comment]
Another example of why we need this functionality - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/PageBlobClientTests.cs#L1057.  It's not programmatically possible to create a managed disk storage account.  That test will never run in live mode.

File: sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:16]: Done.

File: sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:15]: Which constructor parameter are you referring to?

Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:15]: https://github.com/Azure/azure-sdk-for-net/blob/b9bd85cb2591f7c3f3c054f33f158e758056f1d3/sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs#L16

Comment (seanmcc-msft) [reply sdk/core/Azure.Core.TestFramework/src/PlaybackOnlyAttribute.cs:15]: I assume so, yes, but this is by design.  The workflow to re-record would be to comment out the attribute.

**Comment by pakrym** [comment]
@seanmcc-msft what are your thoughts about https://github.com/Azure/azure-sdk-for-net/pull/12663#issuecomment-641543506?

**Comment by seanmcc-msft** [comment]
> @seanmcc-msft what are your thoughts about [#12663 (comment)](https://github.com/Azure/azure-sdk-for-net/pull/12663#issuecomment-641543506)?

I don't think I fully understand that comment.

My objective is to have the ability to have tests that only run in playback mode, both locally and in the pipeline.  These tests would rarely be re-recorded, and never run live.

**Review by pakrym — APPROVED** [review]

**Pull request merged**